python/mp/rosary.py

```diff
@@ -42,47 +43,85 @@ def __init__(self, ip="127.0.0.1", port=5005):
+            'Eighth6': frozenset(self.beads[46:53]),
+            'Eighth7': frozenset(self.beads[53:60]),
+            'Quadrent0': frozenset(self.beads[4:18]),
+            'Quadrent1': frozenset(self.beads[18:32]),
+            'Quadrent2': frozenset(self.beads[32:46]),
+            'Quadrent3': frozenset(self.beads[46:60]),
+            'Even_All': frozenset(self.beads[0:60:2]),
+            'Even_Ring': frozenset(self.beads[4:60:2]),
+            'Odd_All': frozenset(self.beads[1:60:2]),
+            'Odd_Ring': frozenset(self.beads[5:60:2])
+        }
+        self.set_registry['Half01'] = self.set_registry['Quadrent0'].\
+                                           union(self.set_registry['Quadrent1'])
+        self.set_registry['Half12'] = self.set_registry['Quadrent1'].\
+                                           union(self.set_registry['Quadrent2'])
+        self.set_registry['Half23'] = self.set_registry['Quadrent2'].\
+                                           union(self.set_registry['Quadrent3'])
+        self.set_registry['Half30'] = self.set_registry['Quadrent3'].\
+                                           union(self.set_registry['Quadrent0'])
 
         # some useful predefined colors
-        self.Color_White = color.Color(1,1,1)
-        self.Color_Red = color.Color(1,0,0)
-        self.Color_Yellow = color.Color(1,1,0)
-        self.Color_Green = color.Color(0,1,0)
-        self.Color_Blue = color.Color(0,0,1)
-        self.Color_Violet = color.Color(1,0,1)
-        self.Color_Cyan = color.Color(0,1,1)
-        self.Color_Black = color.Color(0,0,0)
+        self.color_registry = {
+            'White': color.Color(1,1,1),
+            'Red': color.Color(1,0,0),
+            'Yellow': color.Color(1,1,0),
+            'Green': color.Color(0,1,0),
+            'Blue': color.Color(0,0,1),
+            'Violet': color.Color(1,0,1),
+            'Cyan': color.Color(0,1,1),
+            'Black': color.Color(0,0,0)
+        }
+
+        # Automagically register effects
+        self.register_defined_effects()
 
     def register_effect(self, effect):
         # instantiate the object so we get get the name
-        e = effect(self.Set_None)
+        e = effect(self.set_registry['None'])
         self.effect_registry[e.name] = effect
+
+    def find_defined_effects(self, module_or_class):
+
+        classes = set()
+
+        # A little imperfect, as we'll first process imports and
+        # we end up trying to add mp.effects.effect.Effect many times
+        # We're circumventing this by using a set
+        # Inspired by:
+        #   http://stackoverflow.com/a/408465
+        #   http://stackoverflow.com/a/22578562
+        for name, obj in inspect.getmembers(module_or_class):
+            if inspect.ismodule(obj) and obj.__package__ == 'mp.effects':
+                classes = classes.union(self.find_defined_effects(obj))
+            elif inspect.isclass(obj):
+                classes.add(obj)
+
+        return classes
```

Comment:
I agree that it doesn't feel as clean, but I don't have any concrete suggestions for improving it either.  Can I ask, what qualifies as a "defined effect", then? And why are we returning a set of classes, rather than, say, a set of effect names or something?